hibernate-vector/src/main/java/org/hibernate/vector/OracleByteVectorJdbcType.java

```diff
@@ -0,0 +1,182 @@
+package org.hibernate.vector;
+
+import java.sql.CallableStatement;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.util.Arrays;
+import java.util.BitSet;
+
+import org.hibernate.dialect.Dialect;
+import org.hibernate.sql.ast.spi.SqlAppender;
+import org.hibernate.type.SqlTypes;
+import org.hibernate.type.descriptor.ValueBinder;
+import org.hibernate.type.descriptor.ValueExtractor;
+import org.hibernate.type.descriptor.WrapperOptions;
+import org.hibernate.type.descriptor.java.JavaType;
+import org.hibernate.type.descriptor.jdbc.ArrayJdbcType;
+import org.hibernate.type.descriptor.jdbc.BasicBinder;
+import org.hibernate.type.descriptor.jdbc.BasicExtractor;
+import org.hibernate.type.descriptor.jdbc.JdbcLiteralFormatter;
+import org.hibernate.type.descriptor.jdbc.JdbcType;
+
+import oracle.jdbc.OracleConnection;
+import oracle.jdbc.OracleType;
+
+/**
+ * Specialized type mapping for single-byte integer vector {@link SqlTypes#VECTOR_INT8} SQL data type for Oracle.
+ *
+ * @author Hassan AL Meftah
+ */
+public class OracleByteVectorJdbcType extends ArrayJdbcType {
+
+
+	private static final byte[] EMPTY = new byte[0];
+
+	public OracleByteVectorJdbcType(JdbcType elementJdbcType) {
+		super( elementJdbcType );
+	}
+
+	@Override
+	public <T> JdbcLiteralFormatter<T> getJdbcLiteralFormatter(JavaType<T> javaTypeDescriptor) {
+		return null;
+	}
+
+	@Override
+	public void appendWriteExpression(String writeExpression, SqlAppender appender, Dialect dialect) {
+		appender.append( "vector(" );
+		appender.append( writeExpression );
+		appender.append( ", *, INT8)" );
+	}
+
+	@Override
+	public String getFriendlyName() {
+		return "VECTOR_INT8";
+	}
+
+	@Override
+	public int getJdbcTypeCode() {
+		return SqlTypes.VECTOR_INT8;
+	}
+
+	@Override
+	public String toString() {
+		return "OracleVectorTypeDescriptor";
+	}
+
+	private boolean isDriverBefore23(WrapperOptions options) throws SQLException {
+		final OracleConnection oracleConnection = options.getSession()
```

Comment:
I'm thinking there could be a cost to acquiring a session. If we want to avoid that cost, we could instead use:
```java
DriverManager.getDriver("jdbc:oracle:thin:").getMajorVersion();
```

Reply:
You should be able to capture this information into `OracleServerConfiguration` and ultimately also in the `OracleDialect`, to then conditionally register respective `JdbcType` implementations.